### DISCUSSION TITLE
gllvm::calc.varpart()?

### DISCUSSION THREAD

**Discussion by gerverska** [JenniNiku/gllvm#175] [Ideas]
Hi everyone!

Having spent some time with `boral`, I was wondering if there were any plans to add a `calc.varpart()` function for `gllvm`. I know there are [concerns to be had](https://stats.stackexchange.com/questions/111150/calculating-r2-in-mixed-models-using-nakagawa-schielzeths-2013-r2glmm-me) in having a feature like this, but I figured I'd inquire all the same.

**Comment by BertvanderVeen** [comment]
Thanks, IIRC it is on the to-do list.

**Reply by gerverska** [reply]
Ah, sorry for doubling up here! Thanks for the heads-up.

**Reply by gerverska** [reply]
I'll also look into how to do this myself--are there any potential pitfalls to be aware of when doing so?

**Reply by BertvanderVeen** [reply]
Probably, though not in the Nakagawa & Schielzeth R^2 sense. You might want to ask Francis or @JenniNiku, they have probably thought more about it than I have.

**Reply by gerverska** [reply]
Man, do you ever take a break on here!? Thanks again as always, Bert.

**Comment by JenniNiku** [comment]
Yes, I'm working on with similar feature

**Reply by gerverska** [reply]
Awesome--I'm excited to hear about this, Jenni!
